lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :address,
+              :collected_fees,
+              :name,
+              :phone,
+              :registered_vehicles, 
+              :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(contact_info)
+    @contact_info = contact_info
+    @address = contact_info[:address]
+    @collected_fees = 0
+    @name = contact_info[:name]
+    @phone = contact_info[:phone]
+    @registered_vehicles = []
     @services = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless @services.include?("Vehicle Registration")
+    
+    if vehicle.antique? 
+      vehicle.assign_plate(:antique) 
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle? 
+      vehicle.assign_plate(:ev)
+      @collected_fees += 200
+    else
+      vehicle.assign_plate(:regular)
+      @collected_fees += 100
+    end
+    
+    @registered_vehicles << vehicle
+    vehicle.time_stamp
+  end
```

Comment:
I like how you have this set up to return nil before even going into the condition.